app/assets/javascripts/discourse/tests/acceptance/keyboard-shortcuts-test.js

```diff
@@ -41,3 +42,67 @@ acceptance("Keyboard Shortcuts", function (needs) {
     assert.equal(currentURL(), "/t/keyboard-shortcuts-are-awesome/27331");
   });
 });
+
+acceptance("Keyboard Shortcuts - Authenticated Users", function (needs) {
+  let resetNewCalled = 0;
+  let markReadCalled = 0;
+  needs.user();
+  needs.pretender((server, helper) => {
+    let topicList = DiscoveryFixtures["/latest.json"];
+
+    // get rid of some of the topics and the more_topics_url
+    // so we consider them allLoaded and show the footer with
+    // the bottom dismiss button
+    topicList.topic_list.topics.splice(20, 30);
+    topicList.topic_list.more_topics_url = null;
+
+    server.get("/unread.json", () => {
+      return helper.response(topicList);
+    });
+    server.get("/new.json", () => {
+      return helper.response(topicList);
+    });
+    server.put("/topics/reset-new", () => {
+      resetNewCalled += 1;
+      return helper.response({});
+    });
+    server.put("/topics/bulk", () => {
+      markReadCalled += 1;
+      return helper.response({});
+    });
+  });
+
+  test("dismiss unread from top and bottom button", async function (assert) {
+    await visit("/unread");
+    await triggerKeyEvent(document, "keypress", "x".charCodeAt(0));
+    await triggerKeyEvent(document, "keypress", "t".charCodeAt(0));
+    assert.ok(exists("#dismiss-read-confirm"));
+    assert.equal(
+      queryAll(".modal-body").text().trim(),
+      "Stop tracking these topics so they never show up as unread for me again"
+    );
+    await click("#dismiss-read-confirm");
+    assert.equal(markReadCalled, 1);
+    document.getElementById("dismiss-topics-top").remove();
```

Comment:
Hmm I'm curious here, why do we have to remove an element from the DOM?

Reply:
So the logic is that the top button only shows up if the bottom button is off the screen when the topics load. So this is a shortcut to avoid having to reload the `unread.json` list with a shorter list of topics. I can do that though if you think it's too hacky?

Reply:
Yea I think it'll be better than altering the DOM in the test suite :)